### PR TITLE
David Kane

### PR DESCRIPTION
**Pull request by DoubleBridges**

### CODE REVIEW

File: assignments/callbacks.js [hunk 1]
```diff
@@ -11,45 +11,88 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
...
+  return cb(arr.length);
 }
 
+let len = len => console.log(len);
+
+getLength(items, len);
+
 function last(arr, cb) {
   // last passes the last item of the array into the callback.
+  return cb(arr[arr.length - 1]);
```
Comment (thepotatohead) [inline assignments/callbacks.js:40]: Good job using the `arr.length - 1` pattern! 🔥

File: assignments/callbacks.js [hunk 2]
```diff
@@ -11,45 +11,88 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
...
 }
 
+multiplyNums(45, 3, (prod) => {
+  console.log(prod)
+});
+
 function contains(item, list, cb) {
   // contains checks if an item is present inside of the given array/list.
   // Pass true to the callback if it is, otherwise pass false.
+  let res = list.includes(item);
```
Comment (thepotatohead) [inline assignments/callbacks.js:68]: I like how you use `includes` to find out if `item` is present in `list`!

File: assignments/callbacks.js [hunk 3]
```diff
@@ -11,45 +11,88 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
...
 /* STRETCH PROBLEM */
 
+const nums = [12, 23, 34, 45, 56, 12, 35, 24, 23, 34, 12];
+
 function removeDuplicates(array, cb) {
   // removeDuplicates removes all duplicate values from the given array.
   // Pass the duplicate free array to the callback function.
   // Do not mutate the original array.
+  let filteredArr = [];
+  array.forEach((val) => {
```
Comment (thepotatohead) [inline assignments/callbacks.js:89]: Awesome job for attempting stretch! You're killing it! 💯

File: assignments/array-methods.js [hunk 4]
```diff
@@ -2,85 +2,124 @@
...
     fullName.push(`${runner.first_name} ${runner.last_name}`);
 });
+
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+let allCaps = [runners.map(runner => runner.first_name.toUpperCase())];
```
Comment (thepotatohead) [inline assignments/array-methods.js:69]: You don't need to wrap around `map` with `[]`. Else, you'll have a nested array as the result 😄

File: assignments/array-methods.js [hunk 5]
```diff
@@ -2,85 +2,124 @@
...
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = [runners.filter(runner => runner.shirt_size === "L")];
+
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = [runners.reduce((acc, runner) => acc + runner.donation, 0)];
```
Comment (thepotatohead) [inline assignments/array-methods.js]: Same here and on other array methods. Most of the array methods that has a loop feature have a return value of an array. So you don't need an extra `[]` 😊

Comment (thepotatohead) [reply assignments/array-methods.js]: And actually in this challenge, the expected result is a number. 😅

Comment (DoubleBridges) [reply assignments/array-methods.js]: I didn't catch that. I think I did catch it on the stretch stuff at the end and I did not think to go back and look for them in the regular assignment

**Pull request merged**